File: DSA Revision/Sid/Arrays/Binary Search.java
class Solution {
    //OM GAN GANAPATHAYE NAMO NAMAH 
    //JAI SHRI RAM 
    //JAI BAJRANGBALI 
    //AMME NARAYANA, DEVI NARAYANA, LAKSHMI NARAYANA, BHADRE NARAYANA
    public int search(int[] nums, int target) {
        int n = nums.length; 
        int l = 0, r = n - 1;
        while(r >= l)
        {
            int mid = (r + l)/2;
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] > target)
                r--;
            else
                l++;
        }
        return -1;
    }
}

File: NeetCode-150/Binary Search/Binary Search.cpp
// Hari

int search(vector<int>& nums, int target) {
        int N = nums.size();
        int l = 0, r = N-1;
        
        while (l <= r) {
            int M = l + ((r - l) / 2);
            if (nums[M] == target) return M;
            else if (nums[M] < target) l = M + 1;
            else r = M - 1;
        }
        
        return -1; 
    }
